src/toolchain/web-guide.md

```diff
@@ -0,0 +1,135 @@
+<!--
+  ~ This Source Code Form is subject to the terms of the Mozilla Public
+  ~ License, v. 2.0. If a copy of the MPL was not distributed with this
+  ~ file, You can obtain one at https://mozilla.org/MPL/2.0/.
+-->
+
+
+# Intro to Web Builds
+
+Web builds are a fair bit more to get started with compared to native builds.
+This will be a complete guide on how to get things compiled.
+Setting up a web server is out-of-scope for this guide, and best found elsewhere.
+
+- Web support with `gdext` is experimental and should be understood as such before proceeding.
+
+---
+
+
+## Installing Pre-Requisites
+
+- Install a nightly build of `rustc`, the `wasm32-unknown-emscripten` target for `rustc`, and `rust-src`.
+Assuming that Rust was installed with rustup, this is quite simple.
+
+  ```sh
+  rustup toolchain install nightly
+  rustup component add rust-src --toolchain nightly
+  rustup target add wasm32-unknown-emscripten --toolchain nightly
+  ```
+
+- Install Emscripten
+  - The simplest way to achieve this is to install `emsdk` from the git repo. Recommended to use version 3.1.39 for now[^1]
+
+  ```sh
+  git clone https://github.com/emscripten-core/emsdk.git
+  cd emsdk
+  ./emsdk install 3.1.39
+  ./emsdk activate 3.1.39
+  source ./emsdk.sh     (or ./emsdk.bat on windows)
+  ```
+
+  - It would also be HIGHLY recommended to follow the instructions in the terminal to add `emcc`[^2] to your `PATH`.
+    If not, it is necessary to manually `source` the `emsdk.sh` file in every new terminal prior to compilation.
+    This is platform-specific.
+
+[^1]: Note: Due to a bug with `emscripten`, the maximum version of `emcc`[^2] that can one compile `Godot` with is `3.1.39`.  `gdext` itself should be able to support the latest version of `emcc`, however, it may be a safer bet to stick to version `3.1.39`.
+[^2]: `emcc` is the name of Emscripten's compiler
+
+---
+
+
+## Project Configuration
+
+- Enable the `experimental-wasm` feature on `gdext` in the `Cargo.toml` file.
+  It is also strongly recommended to enable the `lazy-function-tables` feature to avoid an extremely long compile time with release builds.
+  It is likely to due a bug, but is still being investigated.  Edit the line to something like the following:
+
+  ```toml
+  godot = { git = "https://github.com/godot-rust/gdext", branch = "master", features = ["experimental-wasm", "lazy-function-tables"] }
+  ```
+
+- At the same level as your `Cargo.toml`, create a `.config` directory and inside of that,
+  create a `config.toml` file (if one doesn't already exist). The contents of this file should be as follows
+
+  ```toml
+  [target.wasm32-unknown-emscripten]
+  rustflags = [
+      "-C", "link-args=-sSIDE_MODULE=2",
+      "-C", "link-args=-sUSE_PTHREADS=1",
```

Comment:
Quick remark: it seems that Emscripten, since version 3.1.34, recommends the usage of `-pthread` instead of `-sUSE_PTHREADS=1` (the latter being deprecated <a href="#footnote-1">[1]</a> <a href="#footnote-2">[2]</a>), so we should probably recommend that from the start. @zecozephyr can you confirm everything still works fine with the `-pthread` flag (the two should be equivalent)?

<p id="footnote-1">

[1] See my comment here: https://github.com/godotengine/godot-cpp/pull/1247#issuecomment-1742448769

</p>
<p id="footnote-2">

[2] See the PR to emscripten which made this change: https://github.com/emscripten-core/emscripten/pull/18923

</p>

Reply:
I don't believe there is a rush, but I guess it's better if we adopt the new "good practice" ASAP 🙂

Regarding Godot 4.1.3 being built with Emscripten 3.1.18, you make a point (even though 4.2 is gonna be stable soon); however, I think the usage of -pthread should only depend on your local Emscripten install (so, the one you use to build), not Godot's, so, in theory, it _should_ be fine anyway as we recommend using emsdk 3.1.39 for 4.1.3+ already (so it would be consistent)

Might be worth testing first either way!

Reply:
Is the new flag already supported on our suggested minimum version?

* If yes, I don't see any reason to keep the deprecated one around.
* If no, we should use the deprecated one with a comment, that in newer Emscripten versions, a different flag is used.

Reply:
> Is the new flag already supported on our suggested minimum version?

From the text, the suggested emsdk version seems to be 3.1.39, so yeah.